feat: refactor transitions affecting perf

AdamPawlinski opened the pull request:
# Related issue
Closes #2085 

# Scope of work
Replace transitions with `height`, `top` and `width` to transform property and add `will-change` to `font-size` transitions. 
  

# Screenshots of visual changes


# Checklist

- [x] No commented blocks of code left
- [ ] Run tests and docs
- [x] Self code-reviewed
- [ ] Changes documented 

If applicable:

- [ ] I followed [composition rules](https://docs.storefrontui.io/?path=/story/introduction-contributing-guide-code-guidelines--page) for my component
- [ ] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/vuestorefront/storefront-ui?pullRequest=2183) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/vuestorefront/storefront-ui?pullRequest=2183) before we can accept your contribution.<br/><hr/>**AdamPawlinski** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/vuestorefront/storefront-ui?pullRequest=2183) it.</sub>

aniamusial commented on packages/shared/styles/components/atoms/SfInput.scss:
```diff
@@ -22,14 +22,14 @@
     background: var(--input-label-background);
     transform: var(
       --input-label-transform,
-      translate3d(0, calc(-1 * var(--input-label-top, 50%)), 0)
```
question: isn't this a breaking change, as if someone overrides this ```var(--input-label-top)``` in its project then this will affect the translation?

AdamPawlinski replied:
Yes, it is. I will describe it in migration guide.

aniamusial commented on packages/shared/styles/components/molecules/SfComponentSelect.scss:
```diff
@@ -53,13 +53,14 @@
     color: var(--component-select-label-color, var(--c-secondary-variant));
     transform: var(
       --component-select-label-transform,
-      translate3d(0, calc(-1 * var(--component-select-label-top, 50%)), 0)
```
question: same as above, isn't this a breaking change?

AdamPawlinski replied:
Like above.

AdamPawlinski commented:
I changed nuxt-image config to one object prop `nuxtIgConfig` passed to SfImage, SfProductCard, SfProductCardHorizontal, SfGallery.

AdamPawlinski commented:
I added css custom properties to customize position of label in SfInput, SfComponentSelect, SfSelect after transition, which was not possible after changes applied before. Also I added migration guide with explanation of BCs.

justyna-13 commented on packages/vue/src/utilities/directives/will-change/will-change-directive.js:
```diff
@@ -0,0 +1,18 @@
+  bind(el, binding) {
+    binding.name = "will-change";
+    el._willChangeActive = function () {
+      console.log(binding.value);
```
console.log :)

justyna-13 reviewed (commented):
It works on SfInput and SfComponentSelect after changes but I can't see it on the rest. 
Anyway, I wonder a little bit about this logic - will-change will be added after hovering label element. I think users don't pay attention and click anywhere on input so will-change won't be added in most cases

AdamPawlinski commented on packages/vue/src/components/atoms/SfInput/SfInput.vue:
```diff
@@ -23,7 +24,7 @@
       <label
         :class="{ 'display-none': !label }"
-        class="sf-input__label"
+        class="sf-input__label sf-label"
```
What do you think about changing the name of this class to sth that informs about what it is used for, like`will-change`?

AdamPawlinski replied:
name changed

AdamPawlinski reviewed (commented):
It works perfectly, just one question :)
